cmake/RepositoryDependenciesSetup.cmake

```diff
@@ -10,4 +10,77 @@ if (TPL_ENABLE_CUDA AND NOT Kokkos_ENABLE_Cuda_Relocatable_Device_Code)
   elseif (${PROJECT_NAME}_ENABLE_ShyLU_NodeTacho)
     message(FATAL_ERROR "ERROR: ${PROJECT_NAME}_ENABLE_ShyLU_NodeTacho=ON but TPL_ENABLE_CUDA='${TPL_ENABLE_CUDA}' AND Kokkos_ENABLE_Cuda_Relocatable_Device_Code='${Kokkos_ENABLE_Cuda_Relocatable_Device_Code}' which is not allowed!")
   endif()
-endif() 
\ No newline at end of file
+endif() 
+
+#########################################################################
+# STKBalance does not work with GO=INT or GO=UNSIGNED
+# Warn and disable it if it was set as a dependence of STK
+# Error out if it was explicitly requested by user
+
+IF ("${Tpetra_ENABLE_DEPRECATED_CODE}" STREQUAL "OFF")  # Remove this test when
```

Comment:
I think this needs to be:

```
IF ("${Tpetra_ENABLE_DEPRECATED_CODE}" NOT STREQUAL "" AND NOT Tpetra_ENABLE_DEPRECATED_CODE)
```

Otherwise, if the user sets `-DTpetra_ENABLE_DEPRECATED_CODE=FALSE` (or `NO` or `false`, or any of the string values that CMake recognizes as "false") then it will fail this logic (and confuse users).  See:

* https://tribits.org/doc/TribitsDevelopersGuide.html#how-to-check-for-and-tweak-tribits-enable-cache-variables

Reply:
Good!  Shall do.